docs/proposals/scheduling/20220901-gang-scheduling.md

```diff
@@ -112,13 +112,25 @@ which allow user to bundle different Gangs together.
 It should be noted that, if the resource accumulation conditions of Gang are met, then some pods failed in the process of binding,
 or some bound pods are preempted\rescheduled, should the constraints of Gang still be effective in the process of resource reallocation? 
 Because the initial purpose of Gang is to require pods to be pulled up at the same time, if some pods have been pulled up, 
-then the subsequent Gang behavior is meaningless. Therefore, when once Gang has been satisfied, all subsequent resource allocations 
-are no longer constrained by Gang rules, and their performance is similar to ordinary pod.
+then the subsequent Gang behavior is meaningless. Therefore, once Gang has been satisfied and its match policy is once-satisfied, all 
+subsequent resource allocations are no longer constrained by Gang rules, and their performance is similar to ordinary pod.
 
 As mentioned above, `WaitTime` is the max wait time since first pod comes to permit stage. If `WaitTime` is timeout, 
 scheduler will roll back all assumed pods, update each pod's annotation with `gang.scheduling.koordinator.sh/timeout=true`, and
 won't schedule these pods anymore. User should pay attention to this status and delete pods timely.
 
+#### Gang Match Policy
+When using Gang scheduling, you can expand the logic of determining whether gang is satisfied by `match-policy`. 
```

Comment:
I still don't quite understand the scenes behind this. There are several issues to discuss:
1. From these descriptions, it seems that after minAvailable is satisfied, you want to continue to do Gang Scheduling for subsequent Pods again? 
2. Do these special policies have to be implemented in the scheduler? When the Job Controller (no matter which Job Controller it is) creates  a new batch of Pods, can it not set new Gang parameters for a new batch of Pods?
3. Which of these three policies is the default strategy, and is it compatible with existing implementations?

Reply:
To issue 1, the answer is yes.
To issue 2, if the Job Controller need reconstruct all pods to enhance fault toleration and the pods' reconstruction behavior is to reconstruct pod one by one, Koordinator now will schedule reconstructed pod immediately, which may not meet the requirement of scheduling all pods at the same time to ensure the best performance.
To issue3,as the Koordinator's current implementation is  "once-satisfied" policy, so the default strategy is "once-satisfied" policy to compat with existing implementation.

Reply:
In what situation we have to do Gang for subsequent Pods?
If job's replica is larger than min available, subsequent Pods should pass.
If job's replica is same with min available, why pods come again? If pods failed, the job status will be set as failed for tfjobs.

Reply:
If job's replica is same with min available and pods failed, but the job supports failover, pods will come again.

Reply:
> If job's replica is same with min available and pods failed, but the job supports failover, pods will come again.

If job supports failover, why we should block the subsequent arrival pods? I mean, if all pods should be deleted and created, we can create a new gang, otherwise once-satisfied is enough.

Reply:
Yes, create a new gang is a solution. However, not all workload controller can persist the new gangName, and only-waiting may help.

Reply:
Is there any open source job controller supports failover? How do they support failover and distinguish pods from different rounds？ 
If a job supports failover and all pods should be deleted and created, I think the better way is to include rounds in pod name, this is not difficult. 
@eahydra WDYT